src/installerfomod.cpp

```diff
@@ -84,6 +84,20 @@ bool InstallerFomod::isManualInstaller() const
   return false;
 }
 
+
+void InstallerFomod::onInstallationStart(QString const& archive, bool reinstallation, IModInterface* currentMod)
+{
+  m_InstallerUsed = false;
+}
+
+
+void InstallerFomod::onInstallationEnd(EInstallResult result, IModInterface* newMod)
+{
+  if (result == EInstallResult::RESULT_SUCCESS && m_InstallerUsed) {
+    newMod->setUrl(m_Url);

```

Comment:
The original code didn't overwrite the url of the existing mod in case there was one already. Like if you are reinstalling the mod or merging with the previous one. We want to keep whatever the user had before since it's probably custom.

Reply:
Maybe we want to make it a `GuessedValue` so it can have guess qualities and then metadata would lose to user input, but beat auto-generated data that might have come from elsewhere in MO2.

Reply:
> The original code didn't overwrite the url of the existing mod in case there was one already. Like if you are reinstalling the mod or merging with the previous one. We want to keep whatever the user had before since it's probably custom.

I fixed this to get the old behavior but most of the time there is a previous "custom"URL because it was stored by the previous FOMOD installation, not by the user.

> Maybe we want to make it a GuessedValue so it can have guess qualities and then metadata would lose to user input, but beat auto-generated data that might have come from elsewhere in MO2.

You can do this inside your own installer if you want, but in this case the user cannot modify the URL, it's fetch from `dialog` but only because it's the dialog that read the `.xml` file (I'm not responsible for that design... ) so it's simply the URL in the `.xml` file.

There is no user input in this case, just previously stored value. You could add an input field for each meta-information you want the user and do as we do with the mod name but I don't think that's a good idea from a user perspective.

Furthermore, I'd rather not add parameters to `install()` (in fact I'd rather remove that `nexusId&` parameter at somce point), because you end-up adding parameters after parameters... If you want to set meta-information on mod, you should do it that way now I think.

Reply:
That's not what I was meaning. I meant that we should store some kind of guess quality on-disk so that we can tell whether the user entered the URL manually in the UI or it came from somewhere like the download manager or it came from a previous run of an installer, so either the installer can tell whether it should overwrite what's there or MO2 itself can decide that the new value has lower priority than what's already there and ignore it.

Reply:
I see what you mean, but that requires some (a lot of) changes to the `IModInterface` and the `meta.ini`, especially if we want to apply this for other meta-information, so should probably be a future PR.